src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -651,6 +651,11 @@ public AddBlockResult Insert(BlockHeader header, BlockTreeInsertOptions options
                 throw new InvalidOperationException("Genesis block should not be inserted.");
             }
 
+            if (FindBlock(block.Hash ?? block.CalculateHash(), BlockTreeLookupOptions.TotalDifficultyNotNeeded) != null)
```

Comment:
Do we need this? I think we discussed it and we have something similar in NewPayload